tests/functional/cylc-run/03-remote-run-host.t

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bash
+# THIS FILE IS PART OF THE CYLC SUITE ENGINE.
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#-------------------------------------------------------------------------------
+# Run a workflow with ``cylc run --host=somewhere-else``
+export REQUIRE_PLATFORM='loc:remote fs:shared batch:background'
+. "$(dirname "$0")/test_header"
+set_test_number 2
+
+# shellcheck disable=SC2016
+TEST_DIR="$HOME/cylc-run/" init_suite "${TEST_NAME_BASE}" <<< '
+# A total non-entity workflow - just something to run.
+[scheduling]
+    initial cycle point = 2020
+    [[graph]]
+        R1 = Aleph
+
+[runtime]
+    [[root]]
+        script = cylc stop "${CYLC_SUITE_NAME}" --now --now
```

Comment:
The easy way to do this is to leave the script blank and run the suite with `--no-detach`, this way the grep will never fail due to timing issues.